pkg/testutils/testcluster/testcluster.go

```diff
@@ -1787,6 +1787,7 @@ func (tc *TestCluster) RestartServerWithInspect(
 	// node. This is useful to avoid flakes: the newly restarted node is now on a
 	// different port, and a cycle of gossip is necessary to make all other nodes
 	// aware.
+	id := s.StorageLayer().NodeID()
```

Comment:
Why not just `s.NodeID()`? Is there some kind of mandate to type the "layer" explicitly? CC @knz to help with the answer.

Reply:
We want to be explicit about what we are trying to use StorageLayer (i.e. underlying raw KV) or tenant layer. I think it makes little difference here as we manipulate servers themselves, but maybe having intent expressed explicitly is better. Then we can deprecate those methods on the level above where server could have no storage and no nodeid?